community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -322,8 +325,43 @@ private static DurationValue parseDuration( Matcher matcher )
         {
             return null;
         }
-        long months = optLong( y ) * 12 + optLong( m );
-        long days = optLong( w ) * 7 + optLong( d );
+        int pos;
+        if ( (pos = fractionPoint( y )) >= 0 )
+        {
+            if ( m != null || w != null || d != null || t != null )
+            {
+                return null;
```

Comment:
It turns out that this will actually never happen. The Regex Matcher will already catch these cases. I will add tests for a useful error message, but not change this behavior here.